core/controllers/contributor_dashboard_admin_test.py

```diff
@@ -763,3 +763,831 @@ def test_stats_data_not_calculated_without_topic_id_and_language_code(
             'TranslationContributionStatsDict.',
             response['error']
         )
+
+
+class ContributorDashboardAdminStatsHandlerTest(test_utils.GenericTestBase):
+    """Tests for ContributorDashboardAdminStatsHandler.
+    """
+
+    CONTRIBUTOR_EMAIL: Final = 'contributor@example.com'
+    CONTRIBUTOR_USERNAME: Final = 'contributor'
+    SUGGESTION_LANGUAGE_CODE: Final = 'es'
+    USER_ID_1: Final = 'uid_01234567890123456789012345678912'
+    TOPIC_IDS_WITH_TRANSLATION_SUBMISSIONS: Final = ['topic1', 'topic2']
+    RECENT_REVIEW_OUTCOMES: Final = ['accepted', 'rejected']
+    RECENT_PERFORMANCE: Final = 2
+    OVERALL_ACCURACY: Final = 2.0
+    SUBMITTED_TRANSLATIONS_COUNT: Final = 2
+    SUBMITTED_TRANSLATION_WORD_COUNT: Final = 100
+    TOPIC_IDS_WITH_TRANSLATION_REVIEWS = ['18', '19', '20']
+    REVIEWED_TRANSLATIONS_COUNT = 2
+    ACCEPTED_TRANSLATIONS_COUNT: Final = 1
+    ACCEPTED_TRANSLATIONS_WITHOUT_REVIEWER_EDITS_COUNT: Final = 0
+    ACCEPTED_TRANSLATIONS_WITH_REVIEWER_EDITS_COUNT: Final = 0
+    ACCEPTED_TRANSLATION_WORD_COUNT: Final = 50
+    REJECTED_TRANSLATIONS_COUNT: Final = 0
+    REJECTED_TRANSLATION_WORD_COUNT: Final = 0
+    REVIEWED_QUESTIONS_COUNT = 2
+    ACCEPTED_QUESTIONS_COUNT = 1
+    ACCEPTED_QUESTIONS_WITH_REVIEWER_EDITS_COUNT = 0
+    ACCEPTED_QUESTIONS_WITHOUT_REVIEWER_EDITS_COUNT = 0
+    REJECTED_QUESTIONS_COUNT: Final = 20
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.signup(self.CONTRIBUTOR_EMAIL, self.CONTRIBUTOR_USERNAME)
+        self.contributor_id = self.get_user_id_from_email(
+            self.CONTRIBUTOR_EMAIL)
+        user_services.add_user_role(
+            self.contributor_id, feconf.ROLE_ID_TRANSLATION_ADMIN)
+
+    def test_get_stats_with_invalid_contribution_type_raises_error(
+        self
+    ) -> None:
+        self.login(self.CONTRIBUTOR_EMAIL)
+        csrf_token = self.get_new_csrf_token()
+        response = self.post_json(
+            '/contributor-dashboard-admin-stats/invalid/submission', {
+                'page_size': 2,
+                'offset': 1,
+                'language_code': self.SUGGESTION_LANGUAGE_CODE,
+                'sort_by': None,
+                'topic_ids': None,
+                'num_days_since_last_activity': None
+            },
+                csrf_token=csrf_token,
+                expected_status_int=400)
+
+        self.assertEqual(
+            response['error'], 'Invalid contribution type invalid.')
+
+        self.logout()
```

Comment:
```suggestion
        csrf_token = self.get_new_csrf_token()
        
        response = self.post_json(
            '/contributor-dashboard-admin-stats/invalid/submission', {
                'page_size': 2,
                'offset': 1,
                'language_code': self.SUGGESTION_LANGUAGE_CODE,
                'sort_by': None,
                'topic_ids': None,
                'num_days_since_last_activity': None
            },
                csrf_token=csrf_token,
                expected_status_int=400)

        self.assertEqual(
            response['error'], 'Invalid contribution type invalid.')
        self.logout()
```

Use new lines to differentiate the test structure (setup, action under test, asserts). Ditto below. Also see https://github.com/oppia/oppia/wiki/Backend-tests#guidelines-for-writing-good-tests.

Reply:
Updated. PTAL. Thanks for pointing to the documentation.